metricflow/plan_conversion/dataflow_to_sql.py

```diff
@@ -923,8 +924,31 @@ def visit_combine_metrics_node(self, node: CombineMetricsNode[SqlDataSetT]) -> S
         OUTER JOIN context, because many engines do not support complex ON conditions or other techniques we might
         use to apply a sentinel value for NULL to NULL comparisons.
         """
-        # Sanity check that all parents have the same linkable specs.
-        parent_data_sets = [x.accept(self) for x in node.parent_nodes]
+        parent_data_sets: List[SqlDataSet] = [parent_node.accept(self) for parent_node in node.parent_nodes]
+        uses_time_offset = False
+        input_metric_specs: Dict[str, Sequence[MetricSpec]] = defaultdict(list)
+        parent_aliases_to_data_sets: OrderedDict[str, SqlDataSet] = OrderedDict()
+        table_alias_to_metric_specs: OrderedDict[str, Sequence[MetricSpec]] = OrderedDict()
+        for i, parent_node in enumerate(node.parent_nodes):
+            parent_data_set = parent_data_sets[i]
+            table_alias = self._next_unique_table_alias()
+            parent_data_sets.append(parent_data_set)
+
+            # If metric is derived, get input metrics.
+            if isinstance(parent_node, ComputeMetricsNode):
+                if not uses_time_offset:
+                    uses_time_offset = any(
+                        (metric_spec.offset_window or metric_spec.offset_to_grain_to_date)
+                        for metric_spec in parent_node.metric_specs
+                    )
```

Comment:
Now that we've agreed to support single-measure offsets, is it the case that you need to use a time offset on any metric computation that has that value set? I believe it is, in which case making a separate node for the time spine join and applying the window/offset parameters in that node is likely to be a cleaner approach to covering all of the cases.

Like there's nothing preventing a derived metric from having an offset as an input, or from having two metrics with two different offsets that feed into a derived metric with its own offset. Like you might have some ad revenue conversion thing where you have a conversion event, a billing event, a revenue event, and a finalized revenue event, and those are all running on different offsets to keep them aligned to the conversion date (or to the finalized revenue date, depending).

In that case you'd have a separate TimeSpineJoinNode (or whatever) for each of those offsets, and it'd show up in the DataflowPlan as a formal step. You'd basically be rewriting the metric_time column with the time spine, which is something we need to support anyway, so we can come up with a lightweight extended design that parameterizes the node in the way we think we'll need.